packages/jest-runtime/src/index.ts

```diff
@@ -464,11 +463,7 @@ class Runtime {
     options: InternalModuleOptions | undefined,
     moduleRegistry: ModuleRegistry,
   ) {
-    if (path.extname(modulePath) === '.json') {
-      localModule.exports = this._environment.global.JSON.parse(
-        stripBOM(fs.readFileSync(modulePath, 'utf8')),
```

Comment:
This functionality is still needed, maybe what you need to do is setup a default transformer that does this. It may explain the test failures as well.